qiskit/visualization/matplotlib.py

```diff
@@ -276,16 +226,72 @@ def _registers(self, creg, qreg):
 
     @property
     def ast(self):
+        """AST getter"""
         return self._ast
 
+    def _load_style(self, style):
+        current_style = DefaultStyle().style
+        style_name = 'default'
+        config = user_config.get_config()
+        if style is False:
+            style_name = 'bw'
+        elif isinstance(style, dict) and 'name' in style.keys():
+            style_name = style['name']
+        elif isinstance(style, str):
+            style_name = style
+        elif config:
+            style_name = config.get('circuit_mpl_style', 'default')
+        if style_name[-5:] == '.json':
+            style_name = style_name[:-5]
+
+        # Search for file in 'styles' dir, then config_path, and finally 'cwd'
+        style_path = []
+        if style_name != 'default':
+            style_name = style_name + '.json'
+            dirname = os.path.dirname(__file__)
+            style_path.append(os.path.join(dirname, 'styles', style_name))
+
+            if config:
+                config_path = config.get('circuit_mpl_style_path', '')
+                if config_path:
+                    for path in config_path:
+                        style_path.append(os.path.join(path, style_name))
+            style_path.append(os.path.join('', style_name))
+
+            found_path = False
+            for path in style_path:
+                if os.path.isfile(os.path.expanduser(path)):
+                    found_path = True
+                    try:
+                        with open(os.path.expanduser(path)) as infile:
+                            json_style = json.load(infile)
+                        set_style(current_style, json_style)
+                        break
+                    except FileNotFoundError:
```

Comment:
Line 263 already checks file existance. How this exception possible?

Also, maybe `expanduser` once?